src/_guides/language/language-tour.md

```diff
@@ -1502,9 +1502,15 @@ You can override many of these operators, as described in
 | if null                  | `??`                                           |
 | conditional              | <code><em>expr1</em> ? <em>expr2</em> : <em>expr3</em></code> |
 | cascade                  | `..`                                           |
-| assignment               | `=`    `*=`    `/=`    `~/=`    `%=`    `+=`    `-=`    `<<=`    `>>=`    `&=`    `^=`    `|=`    `??=` |
+| assignment               | `=`    `*=`    `/=`    `+=`    `-=`    `&=`    `^=`    <em>etc.</em> |
 {:.table .table-striped}
 
+<aside class="alert alert-info" markdown="1">
+  **Note:**
+  For the latest, definitive list of operators, consult the
+  [Dart language specification][].
+</aside>
```

Comment:
This is probably not so helpful: A missing operator is a simple issue (you'll just get a syntax error when using it, and it's easy to understand that this operator just can't be used in programs, yet). Moreover, `>>>` is currently in the spec, but implementation was postponed because other things were deemed more critical. So at this point checking the spec won't actually help, in the case of `>>>`; conversely, if developers encounter this kind of problem, it should be easy to clarify the situation (e.g., by pointing to a github issue which requests the implementation).

In contrast, operator precedence is much more subtle: You can have a given expression _e_ and consider what it means. Different precedence rules (say, "`|` has higher precedence than `||`; or maybe it's the other way around?") could then give rise to completely different answers, and the whole point of operator precedence is that the structure which produces those different answers is _invisible_. So the developer who is confused about precedence when writing or reading an expression needs to have a much more complex structure in mind than the developer who's confused about whether or not a given operator exists.

This means that it is not a very good idea to give developers rules about precedence that aren't known to hold. The operator precedence table in the spec is explicitly marked as 'non-normative', and we should be similarly cautious here. To illustrate that the rules aren't just trivially true, consider the following example:

```dart
A()..i = 42..j = 43;
```

If you look in the table you will see that `=` has the lowest precedence and `..` is the second lowest. If you look in the table in the specification you'll see that `=` is right-associative. So we should have the following structure on the above example:

```dart
(A()..i) = ((42..j) = 43);
```

However, that's not the structure you will get from the parser (and it certainly won't do the same thing as the actual program above, it's actually wrong in a whole lot of ways). Here is the actual structure:

```dart
(A()..i = 42)..j = 43;
```

We can't put the parentheses to show that `i = 42` and `j = 43` are subterms of the `..` term, but that's indeed how it is, and if we are to create an abstract syntax tree for `A()..i = 42..j = 43` then the abstract syntax tree for `i = 42` would be a child of one of the `..` nodes, and `j = 43` a child of the other `..` node.

The reason why the reference to the precedence table gave us a completely misleading result is that the precedence table does not completely and correctly reflect the actual grammar of Dart. In particular, `=` occurs as an operator directly in the grammar rule for `<expression>`, but it also occurs in a complex beast that's called `<cascadeSection>`.

A precedence table which is capable of correctly representing this situation _does not exist_. So there is no way we can claim that "here's the precedence table for Dart", because that statement can never be true, no matter how much we'd try to fiddle with that table.

I suppose this whole precedence thing came up (probably many years ago) because somebody asked for it, assuming that the concept would be meaningful for Dart, and then the table got added to the spec.

I'd admit that the table works in a lot of situations, so it may indeed 'be helpful' as the spec says, but in the end it is the grammar that counts.

I cannot add a review comment on a line that has no changes, so I'll put it here: At line 1543, we could add something like the following:

```
<div class="alert alert-warning" markdown="1">
**Warning:**
Operator precedence is an approximation of the behavior of a Dart
parser. For a definitive answer, please consult the grammar.
</div>
```

Finally, it again looks like `[]` should be deleted, in line 1511.